lib/recurly/adyen-token.js

```diff
@@ -0,0 +1,116 @@
+import Emitter from 'component-emitter';
+import errors from './errors';
+
+const debug = require('debug')('recurly:adyen-token');
+
+/**
+ * Instantiation factory
+ *
+ * @param  {Object} options
+ * @return {AdyenToken}
+ */
+export function factory (options) {
+  return new AdyenToken(Object.assign({}, options, { recurly: this }));
+}
+
+/**
+ * Initializes an AdyenToken session.
+ *
+ * @param {Object} options
+ * @param {Recurly} options.recurly
+ * @constructor
+ * @public
+ */
+class AdyenToken extends Emitter {
+  constructor (options) {
+    debug('Creating new Adyen Token session');
+    super();
+    this.recurly = options.recurly;
+  }
+
+  /**
+   * Invokes the AdyenToken tokenization.
+   * @param {Object} opts
+  */
+  token (type, payload) {
+    debug('Invoking AdyenToken tokenization');
+
+    const errors = validatePayload(payload);
+    if (errors.length > 0) {
+      return this.error('validation', { fields: errors });
+    }
+
+    this.recurly.request.post({
+      route: '/adyen_token/token',
+      data: { type, payload },
+      done: (error, token) => {
+        if (error) {
+          return this.error('adyen-token-error', { cause: error });
+        }
+
+        this.emit('token', token);
+      }
+    });
+  }
+
+  error (...params) {
+    const err = params[0] instanceof Error ? params[0] : errors(...params);
+    this.emit('error', err);
+
+    return err;
+  }
+
+  loadIdealBanks ({
+    data: { currency, amount, countryCode } = {},
+    done,
+    attachTo,
+  } = {}) {
+    this.recurly.request.get({
+      route: '/adyen_token/ideal_banks',
+      data: {
```

Comment:
only parameter needed here should be `payment_method_type` and we need to pass in `ideal`